internal/cmd/api-key/command_delete.go

```diff
@@ -28,18 +30,67 @@ func (c *command) newDeleteCommand() *cobra.Command {
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
 	c.setKeyStoreIfNil()
-	apiKey := args[0]
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.ApiKey, apiKey)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
+	if err := c.checkExistence(cmd, args); err != nil {
 		return err
 	}
-	httpResp, err := c.V2Client.DeleteApiKey(apiKey)
 
+	if ok, err := form.ConfirmDeletionYesNo(cmd, resource.ApiKey, args); err != nil || !ok {
+		return err
+	}
+
+	var errs error
+	for _, id := range args {
+		if httpResp, err := c.V2Client.DeleteApiKey(id); err != nil {
+			errs = errors.Join(errs, errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, id, httpResp))
+		} else {
+			output.Printf(errors.DeletedResourceMsg, resource.ApiKey, id)
+			if err := c.keystore.DeleteAPIKey(id); err != nil {
+				errs = errors.Join(errs, err)
+			}
+		}
+	}
+	if errs != nil {
+		errs = errors.NewErrorWithSuggestions(errs.Error(), errors.APIKeyNotFoundSuggestions)
+	}
+
+	return errs
+}
+
+func (c *command) checkExistence(cmd *cobra.Command, args []string) error {
+	// Single
+	if len(args) == 1 {
+		if _, _, err := c.V2Client.GetApiKey(args[0]); err != nil {
+			return errors.NewErrorWithSuggestions(fmt.Sprintf(errors.NotFoundErrorMsg, resource.ApiKey, args[0]), fmt.Sprintf(errors.DeleteNotFoundSuggestions, resource.ApiKey))
+		}
+		return nil
+	}
+
+	// Multiple
+	apiKeys, err := c.V2Client.ListApiKeys("", "")
```

Comment:
What if the user deletes 2 API keys, but they have 1000? This will take a while...

Reply:
I can go ahead and switch to making `n` calls to the describe endpoint instead